ad/movement/trusts.md

```diff
@@ -198,6 +208,83 @@ When doing NTLM authentications across trusts, the trusting domain's domain cont
 
 _Nota bene, wether it's Kerberos or NTLM, the ExtraSids are in the same data structure, it's just named differently for each protocol. And, the SID filtering function called by the trusting DC is the same, for both authentication protocols._
 
+## Bastion/Red Forests
+Microsoft introduced Privileged Access Management (PAM) with Server 2016. Among other things, it has very interesting features like -
+    - A bastion forest (Think the administrative forest in ESAE or the famous Red Forest)
+    - Shadow security principals
+    - Temporary group membership (Add a user to a group with time-to-live (TTL))
+### Overview
+PAM has been discussed in much detail [here by Russel](https://www.petri.com/windows-server-2016-set-privileged-access-management) and [here by Willem](https://blogs.technet.microsoft.com/389thoughts/2017/06/19/ad-2016-pam-trust-how-it-works-and-safety-advisory/). A quick explanation is below:
+1. In a perfect world, PAM enables managing an existing production/user forest using a bastion forest which has a one-way PAM trust with the existing forest. The users in the bastion forest can be 'mapped' to privileges groups like Domain Admins and Enterprise Admins in the user forest without modifying any group memberships or ACLs. This is done by creating Shadow security principals in the bastion forest, which are mapped to SIDs for high privilege groups in the user forest and then add users from the admin forest as members of the shadow security principals.
+2. Shadow Principals reside in a special container 'CN=Shadow Principal Configuration,CN=Services' under the Configuration container on bastion forest.
+3. Now, it is possible to manage powershell.local forest from bastion.local without making any changes in the group memberships or ACLs on powershell.local. This takes away administrative overhead of managing groups and ACLs and reduces chances of lateral movement techniques like OverPTH, PTT and other credential relay techniques.
+4. There is something worth noticing about the above setup. To be able to use the shadow security principals, we had to allow SIDHistory in the PAM trust which means no SID Filtering.
+5. A PAM trust where you do not have an isolated bastion forest is disastrous. Why? Because in such a case if we compromise the bastion forest we get high privileges (Enterprise Admins or Domain Admins) in the other forest:
+    - There is no group membership (unlike Foreign Security Principals)
+    - No ACLs modification
+    - No other modification to look for in the forest which gets compromised!
```

Comment:
The provided content offers a good foundation but can be improved to align more closely with The Hacker Recipes' style and technical guidelines. Here are some suggestions for revision:

1. Title and Structure:
- Replace the "Overview" section title with a more descriptive and specific title, like "Principles of Privileged Access Management (PAM)".

2. Writing Style:
- Shift from a conversational tone to a more formal and passive voice.
- Avoid phrases like "A quick explanation is below" or "There is something worth noticing about the above setup." and directly transition into the explanation.
- Use third-person narrative consistently.

3. Technical Content:
- Start with a brief definition of PAM before diving into details.
- Provide a more detailed explanation of key concepts like "Shadow security principals" and add links to the concepts that were already explained somewhere else (e.g. "SID History" : https://www.thehacker.recipes/ad/movement/trusts#sid-history, same thing for OverPTH, PTT, abusing ACLs, SID Filtering, etc.)

Reply:
Completed. I restructured the whole section, starting with a brief introduction of PAM and then going to the Bastion Forest specifics. Moreover, I changed some phrases to a formal and informative tone per the site's style.

Reply:
I will also add the sources I have found/gotten inspiration from for the new additions because credit should be given to the people that have conducted the original research. Moreover, links to articles that further explain those topics are always a plus.